docs/api/structures/product.md

```diff
@@ -8,4 +8,11 @@
 * `price` number - The cost of the product in the local currency.
 * `formattedPrice` string - The locale formatted price of the product.
 * `currencyCode` string - 3 character code presenting a product's currency based on the ISO 4217 standard.
+* `introductoryPrice` [ProductDiscount](product-discount.md) - The object containing introductory price information for the product.
+available for the product.
+* `discounts` [ProductDiscount](product-discount.md)[] - An array of discount offers
+* `subscriptionGroupIdentifier` string - The identifier of the subscription group to which the subscription belongs.
+* `subscriptionPeriod` [ProductSubscriptionPeriod](product-subscription-period.md) - The period details for products that are subscriptions.
```

Comment:
& same for `ProductSubscriptionPeriod | undefined`.

An empty string is one thing—it can be checked with `if (foo.contentVersion) { /* ... */ }`—but an empty object feels off. Arguably the empty strings aren't a great way to express "not present" either, but it's at least consistent with the existing API.

Reply:
Ah, alright, I'll check the typing again and make objects optional if data within it isn't set.